src/hooks/theme.hook.ts

```diff
@@ -4,20 +4,19 @@ import { ref } from "vue";
 
 interface UseThemeHook {
   themeMode: Ref<EThemeMode>;
-  checkThemeMode: () => EThemeMode;
+  checkThemeMode: () => void;
   changeThemeMode: () => void;
 }
 
 export const useThemeHook = (): UseThemeHook => {
   const themeMode = ref(EThemeMode.LIGHT_MODE);
 
   const checkThemeMode = () => {
-    themeMode.value = localStorage.getItem("themeMode") as EThemeMode;
+    const savedTheme = localStorage.getItem("themeMode") as EThemeMode;
```

Comment:
czemu nie zrobisz tego od razu przy przypisaniu do refa?

Reply:
i nie potrzebujesz ternary w rym wypadku, wystarczy ??

Reply:
Doprecyzuj ten typ lepiej, zamiast asercji użyj wnioskowania typu lub anotacji (najpierw sprawdź wnioskowanie, później anotację).
Niżej sprawdzasz, czy istnieje saved theme więc obecnie asercja jest błędna. Rzutujesz na `EThemeMode`, a to nie jest prawda, bo nie masz pewności czy `themeMode` istnieje w local storage (dlatego sprawdzasz to niżej, co jest jak najbardziej ok ale źle jest to zakomunikowane kompilatorowi, bo myśli że `savedTheme` jest typu `EThemeMode` a nie `EThemeMode | undefined`)